src/web/mjs/connectors/VizShonenJump.mjs

```diff
@@ -34,7 +34,21 @@ export default class VizShonenJump extends Connector {
         };
     }
 
-    async _getMangas() {
+    async _getMangasAvailibleByVolumes() {
+        const request = new Request(new URL('/library', this.url), this.requestOptions);
+        let data = await this.fetchDOM(request);
+
+        data = [...data.querySelectorAll('table.purchase-table a')];
+
+        return data.map(manga => {
+            return {
+                id: manga.pathname,
+                title: manga.innerText.trim()
+            };
+        });
+    }
+
+    async _getMangasAvailibleByChapters() {
         const request = new Request(new URL('/shonenjump', this.url), this.requestOptions);
         let data = await this.fetchDOM(request);
```

Comment:
Consider using the CSS query directly `this.fetchDOM(request, 'div.o_sort_container div.o_sortable a.o_chapters-link')`

Reply:
Wouldn't that kill the geolock check?

Reply:
`data` is the body element, i think a CSS query is missing to find the chapters `this.fetchDOM(request, 'div.mangas')`

Reply:
There is a query to find the chapters: https://github.com/manga-download/hakuneko/blob/d7c1f52adf137762f25a7fed41931fe90731bae3/src/web/mjs/connectors/VizShonenJump.mjs#L55-L72

If I were to implement the fetchDOM with the query, the geolock check wouldn't pass I think right? I'm assuming it'd not contain that text anymore. But I can't test it.

Reply:
Yes you are right, for some reason i didn't see line 59 - 64 during my review ...